pylint/checkers/variables.py

```diff
@@ -2249,6 +2249,22 @@ def _loopvar_name(self, node: astroid.Name) -> None:
         ):
             return
 
+        maybe_walrus = utils.get_node_first_ancestor_of_type(node, nodes.NamedExpr)
+        if maybe_walrus:
+            maybe_comprehension = utils.get_node_first_ancestor_of_type(
+                maybe_walrus, nodes.Comprehension
+            )
+            if maybe_comprehension:
+                comprehension_scope = utils.get_node_first_ancestor_of_type(
+                    maybe_comprehension, nodes.ComprehensionScope
+                )
+                assert comprehension_scope is not None
```

Comment:
This is making me nervous, maybe we can use a safer way to appease mypy ?

Reply:
Not a problem. I wish we had a utility that would alert us if/when these logic assertions fail, so that we could be safe for users and still enlighten ourselves :-)